crates/libs/metadata/src/reader/mod.rs

```diff
@@ -312,14 +314,19 @@ impl<'a> Reader<'a> {
         for _ in 0..named_arg_count {
             let _id = values.read_u8();
             let arg_type = values.read_u8();
-            let name = values.read_str().to_string();
+            let mut name = values.read_str().to_string();
             let arg = match arg_type {
                 0x02 => Value::Bool(values.read_bool()),
                 0x06 => Value::I16(values.read_i16()),
                 0x08 => Value::I32(values.read_i32()),
                 0x09 => Value::U32(values.read_u32()),
                 0x0E => Value::String(values.read_str().to_string()),
                 0x50 => Value::TypeDef(self.get(TypeName::parse(values.read_str())).next().expect("Type not found")),
+                0x55 => {
```

Comment:
Is this being tested anywhere? Perhaps add a test to `crates/tests/metadata` that reads the `SupportedArchitecture` attributes on `SetWindowLongPtrA`.

Reply:
Oh good point, I'll add that. It is being tested in my own code at least ;)

Reply:
So the only attribute that actually hits this code path is `GCPressure` (which I guess is useful to GCs). I've added a test for it but I don't know how reliable that attribute is.

Reply:
As soon as I have the metadata generation completed, we can stop digging around for test cases and just write discrete tests from scratch.